### PR TITLE
Resource servers endpoints

### PR DESCRIPTION
**Pull request by ignaciojonas**

### CODE REVIEW

File: spec/integration/lib/auth0/api/v2/api_users_spec.rb [hunk 1]
```diff
@@ -110,11 +110,12 @@
...
-    it do
-      expect(
-        client.unlink_users_account(primary_user['user_id'], 'auth0', link_user['user_id']).first
-      ).to include('provider' => 'auth0', 'user_id' => primary_user['identities'].first['user_id'])
+
+      it do
+        expect(
+          client.unlink_users_account(primary_user['user_id'], 'auth0', link_user['user_id']).first
+        ).to include('provider' => 'auth0', 'user_id' => primary_user['identities'].first['user_id'])
+      end
```
Comment (leosoubeste) [inline spec/integration/lib/auth0/api/v2/api_users_spec.rb:118]: do you need this change? seems not related to resource_servers endpoint

Comment (ignaciojonas) [inline spec/integration/lib/auth0/api/v2/api_users_spec.rb:118]: It seems we missed to wrap this in the ignore tests. I did it here to have the Build green...

**Comment by leosoubeste** [comment]
Looks good to me

File: lib/auth0/api/v2/resource_servers.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+          fail Auth0::InvalidParameter, 'Must supply a valid resource server id' if resource_server_id.to_s.empty?
+          path = "#{resource_servers_path}/#{resource_server_id}"
+          delete(path)
+        end
+
+        private
+
+        # Resource Servers API path
+        def resource_servers_path
+          @rules_path ||= '/api/v2/resource-servers'
```
Comment (falvarisqueta) [inline lib/auth0/api/v2/resource_servers.rb]: variable name needs to be updated

File: spec/integration/lib/auth0/api/v2/api_resource_servers_spec.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+                                                'signing_secret' => signing_secret,
+                                                'token_lifetime' => token_lifetime)
+    end
+    it do
+      expect(resource_server).to include('name' => name, 'identifier' => identifier, 'signing_alg' => signing_alg,
+                                         'signing_secret' => signing_secret,
+                                         'token_lifetime' => token_lifetime)
+    end
+    it { expect { client.delete_resource_server(resource_server['id']) }.to_not raise_error }
+  end
```
Comment (falvarisqueta) [inline spec/integration/lib/auth0/api/v2/api_resource_servers_spec.rb]: should the fail Auth0::InvalidParameter, 'Must supply a valid resource server id be tested?

File: spec/integration/lib/auth0/api/v2/api_resource_servers_spec.rb [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      client.create_resource_server(identifier, 'name' => name, 'signing_alg' => signing_alg,
+                                                'signing_secret' => signing_secret,
+                                                'token_lifetime' => token_lifetime)
+    end
+    it do
+      expect(resource_server).to include('name' => name, 'identifier' => identifier, 'signing_alg' => signing_alg,
+                                         'signing_secret' => signing_secret,
+                                         'token_lifetime' => token_lifetime)
+    end
+    it { expect { client.delete_resource_server(resource_server['id']) }.to_not raise_error }
```
Comment (falvarisqueta) [inline spec/integration/lib/auth0/api/v2/api_resource_servers_spec.rb:40]: should the fail Auth0::InvalidParameter, 'Must supply a valid resource server id" be tested?

File: spec/lib/auth0/api/v2/resource_servers_spec.rb [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+      @instance.create_resource_server('test', name: 'name',
+                                               signing_alg: 'signing_alg',
+                                               signing_secret: 'signing_secret',
+                                               token_lifetime: 'token_lifetime',
+                                               scopes: 'scopes')
+    end
+    it 'expect to raise an error when calling with empty identifier' do
+      expect { @instance.create_resource_server(nil) }.to raise_error 'Must supply a valid resource server id'
+    end
```
Comment (falvarisqueta) [inline spec/lib/auth0/api/v2/resource_servers_spec.rb:41]: should we test creating a resource server with name containing "<". It should return error as the character is not supported

File: spec/integration/lib/auth0/api/v2/api_resource_servers_spec.rb (hunk 4 above)
Comment (ignaciojonas) [inline spec/integration/lib/auth0/api/v2/api_resource_servers_spec.rb:40]: This case is covered in the Unit Tests

File: lib/auth0/api/v2/resource_servers.rb (hunk 2 above)
Comment (ignaciojonas) [inline lib/auth0/api/v2/resource_servers.rb]: good catch! :)

File: spec/lib/auth0/api/v2/resource_servers_spec.rb (hunk 5 above)
Comment (ignaciojonas) [inline spec/lib/auth0/api/v2/resource_servers_spec.rb:41]: You are right, I'll add the test to cover this scenario.

File: spec/integration/lib/auth0/api/v2/api_resource_servers_spec.rb (hunk 4 above)
Comment (falvarisqueta) [inline spec/integration/lib/auth0/api/v2/api_resource_servers_spec.rb:40]: the reason i asked is because delete_resource_server is testing the Invalid parameter error in the unit test and here as well (line 45). Should we be consistent where those scenarios are covered?

File: spec/lib/auth0/api/v2/resource_servers_spec.rb [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    it 'expect to raise an error when name contains < or > characters' do
+      expect { @instance.create_resource_server('test', name: '<') }.to raise_error(
+        'Name must contain at least one character. Does not allow "<" or ">"')
+      expect { @instance.create_resource_server('test', name: '<') }.to raise_error(
```
Comment (leosoubeste) [inline spec/lib/auth0/api/v2/resource_servers_spec.rb]: Test is duplicated. Should second one have '>' sign?

File: spec/integration/lib/auth0/api/v2/api_resource_servers_spec.rb (hunk 4 above)
Comment (ignaciojonas) [inline spec/integration/lib/auth0/api/v2/api_resource_servers_spec.rb:40]: You are right!... I removed the integration test for that scenario. It's already covered in Unit Tests.

**Comment by falvarisqueta** [comment]
Looks good to me
